Framework/PythonInterface/mantid/api/src/Exports/AlgorithmManager.cpp

```diff
@@ -43,8 +43,9 @@ AlgorithmManagerImpl &instance() {
 }
 
 IAlgorithm_sptr create(AlgorithmManagerImpl *self, const std::string &algName, const int &version = -1) {
+  self->removeFinishedAlgorithms();
```

Comment:
The comment above this function in the AlgorithmManager suggests that it assumes the mutex has been locked before this gets called. Might I suggest that we lock a mutex at the top of the `removeFinishedAlgorithms` function so that we ensure we don't get any race conditions?

It might also be a good idea to refactor the mutex's in this file to use a `std::recursive_mutex`. This will prevent us from accidentally causing a deadlock by locking the same mutex twice. (To provide an example, the same is done in this PR #36224)

Reply:
Ok